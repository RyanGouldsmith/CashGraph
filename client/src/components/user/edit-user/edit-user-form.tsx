import React from 'react';

import { MutationFn } from 'react-apollo';

interface EditFormProps {
  submitForm: Function;
  editUser: MutationFn;
  userId: String;
  emailRef: React.Ref<HTMLInputElement>;
  nameRef: React.Ref<HTMLInputElement>;
  userData: {
    name: string;
    email: string;
  };
}

export const EditUserForm: React.SFC<EditFormProps> = ({
  submitForm,
  editUser,
  userId,
  emailRef,
  nameRef,
  userData,
}) => (
  <form onSubmit={submitForm(editUser, userId)}>
    <label htmlFor="name">Edit your name</label>
    <input id="name" type="text" defaultValue={userData.name} ref={nameRef} />
    <label htmlFor="email">Edit your email</label>
    <input id="email" type="email" defaultValue={userData.email} ref={emailRef} />
    <button type="submit">Edit Details</button>
  </form>
);
