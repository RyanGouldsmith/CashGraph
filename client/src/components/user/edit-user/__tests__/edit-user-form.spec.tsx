import 'babel-polyfill';
import React from 'react';
import { render } from 'react-testing-library';
import { MutationFn } from 'react-apollo';

import { EditUserForm } from '../edit-user-form';

test('should edit the information and update the users details', async () => {
  const editUser: MutationFn = () => new Promise(resolve => resolve());

  const props = {
    submitForm: () => {},
    editUser,
    userId: '123',
    emailRef: React.createRef<HTMLInputElement>(),
    nameRef: React.createRef<HTMLInputElement>(),
    userData: {
      name: 'testName',
      email: 'testEmail',
    },
  };

  expect(render(<EditUserForm {...props} />)).toMatchSnapshot();
});
