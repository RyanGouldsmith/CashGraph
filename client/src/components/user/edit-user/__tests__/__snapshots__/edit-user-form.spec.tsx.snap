// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should edit the information and update the users details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <label
          for="name"
        >
          Edit your name
        </label>
        <input
          id="name"
          type="text"
          value="testName"
        />
        <label
          for="email"
        >
          Edit your email
        </label>
        <input
          id="email"
          type="email"
          value="testEmail"
        />
        <button
          type="submit"
        >
          Edit Details
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <label
        for="name"
      >
        Edit your name
      </label>
      <input
        id="name"
        type="text"
        value="testName"
      />
      <label
        for="email"
      >
        Edit your email
      </label>
      <input
        id="email"
        type="email"
        value="testEmail"
      />
      <button
        type="submit"
      >
        Edit Details
      </button>
    </form>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
